backup_mongo.sh

```diff
@@ -51,26 +52,41 @@ elif [ -z "$MONGO_USERNAME" ] || [ -z "$MONGO_PASSWORD" ] || [ -z "$MONGO_AUTH_D
   exit 5
 fi
 
+DB_ARG="--db ${DB}"
+if [ "${DB}" = "." ] || [ "${DB}" = "*" ] || [ "${DB}" = "all" ]; then
+  DB=all
+  DB_ARG=
+fi
+
 DIRECTORY=$(date +%Y-%m-%d)
 
-BACKUP_NAME=${DB}-$(date +%Y%m%d_%H%M%S).gz
+BACKUP_NAME="${DB}-$(date +%Y%m%d_%H%M%S).gz"
+
+# if prefix is enabled include the mongo uri in the backup name
+if [ ! -z "${USE_MONGO_URI_PREFIX}" ]; then
+  BACKUP_NAME_PREFIX="${MONGO_URI//[:]/-}-"
+fi
+
+LOCAL_PATH="$HOME/tmp_dump.gz"
+REMOTE_PATH="https://${AZURE_SA}.${AZURE_TYPE}.core.windows.net/${AZURE_CONTAINER_NAME}/${DIRECTORY}/${BACKUP_NAME_PREFIX}${BACKUP_NAME}"
+REMOTE_LATEST_PATH="https://${AZURE_SA}.${AZURE_TYPE}.core.windows.net/${AZURE_CONTAINER_NAME}/latest/${BACKUP_NAME_PREFIX}${DB}-backup.gz"
 
 date
-echo "Backing up MongoDB database ${DB}"
+echo "Backing up MongoDB database(s) ${DB}"
 
-echo "Dumping MongoDB $DB database to compressed archive"
+echo "Dumping MongoDB $DB database(s) to compressed archive"
 if [ "$NO_AUTH" = true ]
 then
-  mongodump --host ${MONGO_URI} --db ${DB} --archive=$HOME/tmp_dump.gz --gzip
+  mongodump --host "${MONGO_URI}" ${DB_ARG} --archive="${LOCAL_PATH}" --gzip
 else
-  mongodump --authenticationDatabase ${MONGO_AUTH_DB} -u ${MONGO_USERNAME} -p ${MONGO_PASSWORD} --host ${MONGO_URI} --db ${DB} --archive=$HOME/tmp_dump.gz --gzip
+  mongodump --authenticationDatabase "${MONGO_AUTH_DB}" -u "${MONGO_USERNAME}" -p "${MONGO_PASSWORD}" --host "${MONGO_URI}" ${DB_ARG} --archive="${LOCAL_PATH}" --gzip
 fi
 
-echo "Copying compressed archive to Azure Storage: ${AZURE_SA}.${AZURE_TYPE}/${AZURE_CONTAINER_NAME}/${DIRECTORY}/${BACKUP_NAME}"
-azcopy --source $HOME/tmp_dump.gz --destination https://${AZURE_SA}.${AZURE_TYPE}.core.windows.net/${AZURE_CONTAINER_NAME}/${DIRECTORY}/${BACKUP_NAME} --dest-key ${AZURE_DESTINATION_KEY}
-yes | azcopy --source $HOME/tmp_dump.gz --destination https://${AZURE_SA}.${AZURE_TYPE}.core.windows.net/${AZURE_CONTAINER_NAME}/latest/${DB}-backup.gz --dest-key ${AZURE_DESTINATION_KEY}
+echo "Copying compressed archive to Azure Storage: ${AZURE_SA}.${AZURE_TYPE}/${AZURE_CONTAINER_NAME}/${DIRECTORY}/${BACKUP_NAME_PREFIX}${BACKUP_NAME}"
+azcopy --source "${LOCAL_PATH}" --destination "${REMOTE_PATH}" --dest-key "${AZURE_DESTINATION_KEY}"
+yes | azcopy --source "${REMOTE_PATH}" --destination "${REMOTE_LATEST_PATH}" --source-key "${AZURE_DESTINATION_KEY}" --dest-key "${AZURE_DESTINATION_KEY}"
```

Comment:
Here we ask Azure to copy the file entire at the remote side - this is much faster

Reply:
TBH I've not tested this in blob mode